Makefile

```diff
@@ -8,44 +8,65 @@ MOCK_FLAG =
 SOURCE_F = sources
 TEST_F = tests
 
+COMMANDS_NAMES = option_cd.c option_echo.c option_env.c option_exit.c option_export.c option_external.c option_pwd.c option_unset.c get_envvars_for_execve.c
+COMMANDS_F = commands
+COMMANDS_SRCS = $(addprefix $(COMMANDS_F)/,$(COMMANDS_NAMES))
+
 INPUT_TO_TEXT_TREE_MOCK_NAMES = input_to_text_tree_mock.c mock_0.c mock_1.c mock_2.c mock_3.c mock_4.c mock_5.c mock_6.c mock_7.c
 INPUT_TO_TEXT_TREE_MOCK_F = input_to_text_tree_mocks
 INPUT_TO_TEXT_TREE_MOCK_SRCS = $(addprefix $(INPUT_TO_TEXT_TREE_MOCK_F)/, $(INPUT_TO_TEXT_TREE_MOCK_NAMES))
 
-COMMANDS_NAMES = option_cd.c option_echo.c option_env.c option_exit.c option_export.c option_external.c option_pwd.c option_unset.c get_envvars_for_execve.c
-COMMANDS_F = commands
-COMMANDS_SRCS = $(addprefix $(COMMANDS_F)/,$(COMMANDS_NAMES))
+PARSING_NAMES = parse_command.c parse_commands_utils.c
+PARSING_F = parsing
+PARSING_SRCS = $(addprefix $(PARSING_F)/, $(PARSING_NAMES))
+
+TREE_CREATION_NAMES = input_to_text_tree_mock.c input_to_text_tree.c
+TREE_CREATION_F = tree_creation
+TREE_CREATION_SRCS = $(addprefix $(TREE_CREATION_F)/, $(TREE_CREATION_NAMES))
 
-SRC_NAMES = finalize.c param_init.c param_get_envvars.c wrappers.c input_to_text_tree.c $(INPUT_TO_TEXT_TREE_MOCK_SRCS) exec_text_tree.c exec_text_tree_node.c $(COMMANDS_SRCS)
+SRC_NAMES = finalize.c param_init.c param_get_envvars.c wrappers.c input_to_text_tree.c $(INPUT_TO_TEXT_TREE_MOCK_SRCS) exec_text_tree.c exec_text_tree_node.c $(COMMANDS_SRCS) $(PARSING_SRCS)
 ENDPOINT_NAME = main.c
 
 SRC_SRCS = $(addprefix $(SOURCE_F)/, $(SRC_NAMES))
 ENDPOINT_SRC = $(addprefix $(SOURCE_F)/, $(ENDPOINT_NAME))
 
-OBJS = $(SRC_SRCS:.c=.o)
-ENDPOINT_OBJ = $(ENDPOINT_SRC:.c=.o)
+OBJ_F = objects/
```

Comment:
Merging several features in a single PR is a VERY bad practice.

1. the more code you are merging, the harder it is to locate errors. On the other hand, if you are merging a small change, and the error arises there, then you can fix it before implementing bigger changes
2. it's harder to review, I have to keep several tracks of your thought in mind. Imagine if I present you EVERYTHING I can make in a week (like three features) - what are your chances to understand what I'm merging?